pkg/task/meta.go

```diff
@@ -26,137 +26,210 @@ import (
+	ListTA MetaTaskAction = "list"
 	// PutTA flags a action as put. Typically used to put
 	// an object.
-	PutTA TaskAction = "put"
+	PutTA MetaTaskAction = "put"
 	// DeleteTA flags a action as delete. Typically used to
 	// delete an object.
-	DeleteTA TaskAction = "delete"
+	DeleteTA MetaTaskAction = "delete"
 	// PatchTA flags a action as patch. Typically used to
 	// patch an object.
-	PatchTA TaskAction = "patch"
+	PatchTA MetaTaskAction = "patch"
 	// OutputTA flags the task action as output. Typically used to
 	// provide a schema (i.e. a custom defined) based output after
 	// running one or more tasks.
-	OutputTA TaskAction = "output"
+	OutputTA MetaTaskAction = "output"
 )
 
-// MetaTask contains information about a Task
-type MetaTask struct {
-	// TaskIdentity provides the required identity
-	// about the task
-	TaskIdentity
+// MetaTaskProps provides properties representing the task's meta
+// information
+type MetaTaskProps struct {
 	// RunNamespace is the namespace where task will get
 	// executed
 	RunNamespace string `json:"runNamespace"`
-	// Action to be invoked on the task
-	Action TaskAction `json:"action"`
 	// Owner represents the owner of this task
 	Owner string `json:"owner"`
-	// ObjectName is the name of the target that is
-	// created or operated by the task
+	// ObjectName is the name of the resource that gets
+	// created or will get operated by this task
 	ObjectName string `json:"objectName"`
 	// Options is a set of selectors that can be used for
-	// get or list actions
+	// tasks that are get or list based actions
 	Options string `json:"options"`
-	// @Deprecated
-	//  In favour of 'RunTask.post' property
-	//
-	// TaskResultQueries will consist of the queries to be run against the
-	// task's result
-	TaskResultQueries []TaskResultQuery `json:"queries"`
-	// Retry specifies the no. of times this task can be tried
-	// This is typically used along with task result verify options
-	// for get or list related actions
+	// Retry specifies the no. of times this same task (i.e. all proeprties
+	// remains same) can be re-tried. This is typically used along with task
+	// result verify options for get or list related actions
 	//
 	// A sample retry option:
 	//
 	// # max of 10 attempts in 20 seconds interval
 	// retry: "10,20s"
 	Retry string `json:"retry"`
-	// RepeatWith sets one or more resources of a given kind that leads to
-	// repetitive execution of a task (in other words a task template is executed
-	// multiple times). The repetition count is determined by the resource names
-	// specified. Each task execution is based on a resource from this provided
-	// list of resources.
+}
+
+// toString returns a string representation of MetaTaskProps structure. In this
+// representation, each property is separated from its value via '='. In
+// addition each property=value pair is separated from other pair via '::'.
+//
+// Example:
+//  runNamespace=default::objectName=MySvc::retry=3,20s
```

Comment:
Any specific reason that we are using `::` as separator? Why not `;` or `,` or '|'.

Reply:
This toString is the string representation of the entire struct & is used to set the current repeat resource.

i.e. in pkg/task/task.go
```go
// set the currently active repeat item
current, err = m.metaTaskExec.repeater.getItem(idx)
```

Anything that can be a good unique delimiter can be used.

Reply:
Need to verify if `.ListItems.currentRepeatResource` is used appropriately in run task templating.